habitat_baselines/common/base_trainer.py

```diff
@@ -130,6 +152,18 @@ def eval(self) -> None:
                         checkpoint_index=prev_ckpt_ind,
                     )
 
+                    save_resume_state(
```

Comment:
Why are we saving the state in the evaluation? Feels kind of confusing to me. If it is needed, can you add a comment as of why?